go/tasks/plugins/k8s/sidecar/sidecar.go

```diff
@@ -73,45 +73,54 @@ type sidecarJob struct {
 	PrimaryContainerName string
 }
 
-func unmarshalSidecarCustom(structObj *structpb.Struct, sidecarJob *sidecarJob) error {
-	if structObj == nil {
-		return fmt.Errorf("nil Struct Object passed")
-	}
-
-	jsonObj, err := json.Marshal(structObj)
-	if err != nil {
-		return err
-	}
-
-	if err = json.Unmarshal(jsonObj, sidecarJob); err != nil {
-		return err
-	}
-
-	return nil
-}
-
 func (sidecarResourceHandler) BuildResource(ctx context.Context, taskCtx pluginsCore.TaskExecutionContext) (client.Object, error) {
-	sidecarJob := sidecarJob{}
+	var podSpec k8sv1.PodSpec
```

Comment:
nit: maybe make this a pointer, and just assign to it in the if and else below, then have the check `if podSpec == nil` outside the if taskTypeVersion condition?

Reply:
I can't make it a pointer! unmarshaling the struct into a nil pointer fails unfortunately https://golang.org/pkg/encoding/json/#InvalidUnmarshalError

Reply:
ah right 👍